x-pack/metricbeat/module/aws/sqs/sqs.go

```diff
@@ -0,0 +1,194 @@
+		}
+
+		metricDataQueries := constructMetricQueries(listMetricsOutput, int64(m.PeriodInSec))
+		if len(metricDataQueries) == 0 {
+			continue
+		}
+
+		init := true
+		getMetricDataOutput := &cloudwatch.GetMetricDataOutput{NextToken: nil}
+		for init || getMetricDataOutput.NextToken != nil {
+			init = false
+			output, err := aws.GetMetricDataPerRegion(metricDataQueries, getMetricDataOutput.NextToken, svcCloudwatch, startTime, endTime)
+			if err != nil {
+				err = errors.Wrap(err, "getMetricDataPerRegion failed, skipping region "+regionName)
+				m.logger.Error(err.Error())
+				report.Error(err)
+				continue
+			}
+			getMetricDataOutput.MetricDataResults = append(getMetricDataOutput.MetricDataResults, output.MetricDataResults...)
+		}
+
+		// Create Cloudwatch Events for SQS
+		event, err := createSQSEvents(getMetricDataOutput.MetricDataResults, metricsetName, regionName, schemaRequestFields)
+		if err != nil {
+			m.logger.Error(err.Error())
+			event.Error = err
+			report.Event(event)
+			continue
+		}
+
+		report.Event(event)
+	}
+}
+
+func constructMetricQueries(listMetricsOutput []cloudwatch.Metric, period int64) []cloudwatch.MetricDataQuery {
+	metricDataQueries := []cloudwatch.MetricDataQuery{}
+	for i, listMetric := range listMetricsOutput {
+		metricDataQuery := createMetricDataQuery(listMetric, i, period)
+		metricDataQueries = append(metricDataQueries, metricDataQuery)
+	}
+	return metricDataQueries
+}
+
+func createMetricDataQuery(metric cloudwatch.Metric, index int, period int64) (metricDataQuery cloudwatch.MetricDataQuery) {
+	statistic := "Average"
+	id := "sqs" + strconv.Itoa(index)
+	metricDims := metric.Dimensions
+	metricName := *metric.MetricName
+	queueName := ""
+	for _, dim := range metricDims {
+		if *dim.Name == "QueueName" {
+			queueName = *dim.Value
+		}
+	}
+	label := queueName + " " + metricName
+
+	metricDataQuery = cloudwatch.MetricDataQuery{
+		Id: &id,
+		MetricStat: &cloudwatch.MetricStat{
+			Period: &period,
+			Stat:   &statistic,
+			Metric: &metric,
+		},
+		Label: &label,
+	}
+	return
+}
+
+func createSQSEvents(getMetricDataResults []cloudwatch.MetricDataResult, metricsetName string, regionName string, schemaMetricFields s.Schema) (event mb.Event, err error) {
+	event.Service = metricsetName
+	event.RootFields = common.MapStr{}
+	event.RootFields.Put("service.name", metricsetName)
+	event.RootFields.Put("cloud.region", regionName)
+
+	mapOfMetricSetFieldResults := make(map[string]interface{})
+	for _, output := range getMetricDataResults {
+		if len(output.Values) == 0 {
+			continue
+		}
+		labels := strings.Split(*output.Label, " ")
+		mapOfMetricSetFieldResults["queue.name"] = labels[0]
+		mapOfMetricSetFieldResults[labels[1]] = fmt.Sprint(output.Values[0])
+	}
+
+	resultMetricSetFields, err := aws.EventMapping(mapOfMetricSetFieldResults, schemaMetricFields)
+	if err != nil {
+		err = errors.Wrap(err, "Error trying to apply schema schemaMetricSetFields in AWS S3 metricbeat module.")
```

Comment:
Typo? `schema schemaMetricSetFields`?

Reply:
yes :-) and should be AWS SQS instead of S3... Thanks!